compiler/qsc_eval/src/lib.rs

```diff
@@ -1108,8 +1201,10 @@ impl State {
                 let scope = env.0.last_mut().expect("binding should have a scope");
                 match scope.bindings.entry(variable.id) {
                     Entry::Vacant(entry) => entry.insert(Variable {
+                        name: variable.name.clone(),
```

Comment:
Food for thought: right now the type string displayed in the debugger comes from the value, which will just give you the top level type (ie: `Tuple` instead of `(Int, Qubit[])`). If you wanted the full type you could get it here with something like `ty_name: pat.ty.to_string(),` and then bubbling that up from the locals instead of using the `value.ty_name()`. That would cause an allocation of the string on each binding, which would have a perf impact, so not sure if there is a better way to accomplish that of if the benefit would be worth the cost.